python/pip_install/parse_requirements_to_bzl/extract_single_wheel/__init__.py

```diff
@@ -29,15 +30,23 @@ def main() -> None:
     if args.extra_pip_args:
         pip_args += json.loads(args.extra_pip_args)["args"]
 
-    with NamedTemporaryFile(mode='wb') as requirement_file:
+    requirement_file = NamedTemporaryFile(mode='wb', delete=False)
```

Comment:
> This is due to Python holding a handle to the temporary file preventing pip, which is run as a sub-process, from reading it.

Interesting. This looks to contain elaboration of the issue https://bugs.python.org/issue14243